client/state/selectors/get-blocked-sites.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * External dependencies
+ */
+import { map, pickBy } from 'lodash';
+
+/**
+ * Returns a list of site IDs blocked by the user
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {Array}        Blocked site IDs
+ */
+export default function getBlockedSites( state ) {
+	return map( Object.keys( pickBy( state.reader.siteBlocks.items ) ), Number );
```

Comment:
i was less concerned about the extra true  in `{ [key]:true }` and more concerned about O(n) lookup time for arrays.  I'll look into whether there are enough sites for it to matter.